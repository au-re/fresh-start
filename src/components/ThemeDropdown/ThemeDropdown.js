import { Popover } from "antd";
import get from "lodash.get";
import React, { Component, Fragment } from "react";
import styled from "styled-components";

import { themeWrapper } from "../../styles/ThemeProvider/ThemeProvider";

const DropdownItem = styled.div`
  min-width: 120px;
  color: ${(props) => (props.isSelected ? get(props, "theme.palette.primary") : "#333")};
  font-size: 1rem;
  cursor: pointer;

  :hover {
    color:
  }
`;

const DropdownMenu = themeWrapper(({ themes, setTheme, selectedTheme }) => (
  <Fragment>
    {
      themes.map((theme, idx) =>
        <DropdownItem
          key={idx}
          onClick={() => setTheme(theme)}
          isSelected={selectedTheme.name === theme.name}>
          {theme.name}
        </DropdownItem>)
    }
  </Fragment>));

const Button = styled.div`
    opacity: 0.8;
    transition: all 0.2s ease-in-out;
    cursor: pointer;
    text-decoration: underline dashed white;
    min-width: 120px;
    color: ${(props) => get(props, "theme.blends.menu.text")};

    :hover {
      opacity: 1;
    }
`;

class ThemeDropdown extends Component {

  static Button = Button;

  render() {
    const { children, ...rest } = this.props;
    return (
      <Popover content={<DropdownMenu />} {...rest}>
        {children}
      </Popover>);
  }
}

export default ThemeDropdown;
