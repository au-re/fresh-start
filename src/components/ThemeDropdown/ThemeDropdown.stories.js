import { withInfo } from "@storybook/addon-info";
import { storiesOf } from "@storybook/react";
import React from "react";

import ThemeProvider from "../../styles/ThemeProvider/ThemeProvider";
import ThemeDropdown from "./ThemeDropdown";

storiesOf("ThemeDropdown", module)

  .add("basic usage", withInfo(`Dropdown for changing the theme`)(
    () =>
      <ThemeProvider>
        <ThemeDropdown trigger="click" placement="bottomRight">
          <ThemeDropdown.Button style={{ position: "fixed", color: "#333" }}>
            change theme
        </ThemeDropdown.Button>
        </ThemeDropdown>
      </ThemeProvider>));
