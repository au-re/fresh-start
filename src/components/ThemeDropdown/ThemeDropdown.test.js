import "jest-styled-components";

import React from "react";
import renderer from "react-test-renderer";

import ThemeDropdown from "./ThemeDropdown";

describe("ThemeDropdown", () => {

  it("renders without props", () => {
    const tree = renderer.create(<ThemeDropdown />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
