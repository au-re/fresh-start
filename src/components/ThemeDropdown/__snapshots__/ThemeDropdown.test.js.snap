// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThemeDropdown renders without props 1`] = `
<span
  onBlur={undefined}
  onClick={undefined}
  onContextMenu={undefined}
  onFocus={undefined}
  onMouseDown={undefined}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onTouchStart={undefined}
/>
`;
